internal/server.go

```diff
@@ -51,7 +52,19 @@ func (s *Server) RootHandler(w http.ResponseWriter, r *http.Request) {
 	// Modify request
 	r.Method = r.Header.Get("X-Forwarded-Method")
 	r.Host = r.Header.Get("X-Forwarded-Host")
-	r.URL, _ = url.Parse(r.Header.Get("X-Forwarded-Uri"))
+
+	path := r.Header.Get("X-Forwarded-Uri")
+	r.URL, _ = url.Parse(path)
+
+	prefix := r.Header.Get("X-Forwarded-Prefix")
+	if prefix != "" {
+		// Use X-Forwarded-Prefix if present (PathPrefixStrip traefik rules)
+		r.URL, _ = url.Parse(fmt.Sprintf("%s%s", prefix, path))
+	}
+	// Use X-Replaced-Path if present (Traefik Modifiers)
+	if r.Header.Get("X-Replaced-Path") != "" {
+		r.URL, _ = url.Parse(r.Header.Get("X-Replaced-Path"))
+	}
```

Comment:
I think a cleaner approach to this might be to extract and modify the `path` variable, and then run url.Parse once on the final path.
Also, it might be better to run the "X-Replaced-Path" check first, then we can have an `else if` before running the "X-Forwarded-Prefix" check?

Reply:
Hi,  something like that ?
```go
func (s *Server) RootHandler(w http.ResponseWriter, r *http.Request) {
	// Modify request
	r.Method = r.Header.Get("X-Forwarded-Method")
	r.Host = r.Header.Get("X-Forwarded-Host")

	// Use X-Replaced-Path if present (Traefik Modifiers)
	if r.Header.Get("X-Replaced-Path") != "" {
		r.URL, _ = url.Parse(r.Header.Get("X-Replaced-Path"))
	} else {
		path := r.Header.Get("X-Forwarded-Uri")
		r.URL, _ = url.Parse(path)
		prefix := r.Header.Get("X-Forwarded-Prefix")
		if prefix != "" {
			// Use X-Forwarded-Prefix if present (PathPrefixStrip traefik rules)
			r.URL, _ = url.Parse(fmt.Sprintf("%s%s", prefix, path))
		}
	}

	// Pass to mux
	s.router.ServeHTTP(w, r)
}
```